warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1307,30 +1332,49 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
                     // predicates based on cost...
                     config.setExpandAllTerms(true);
                     
-                    queryTree = (ASTJexlScript) regexExpansion.visit(queryTree, null);
-                    if (log.isDebugEnabled()) {
-                        logQuery(queryTree, "Query after expanding regex again:");
+                    nodeCount = NodeTypeCountVisitor.countNodes(queryTree);
+                    // Check if there is any regex to expand.
+                    if (nodeCount.hasAny(ASTNRNode.class, ASTERNode.class)) {
+                        queryTree = (ASTJexlScript) regexExpansion.visit(queryTree, null);
+                        if (log.isDebugEnabled()) {
+                            logQuery(queryTree, "Query after expanding regex again:");
+                        }
+                        // The query tree was modified. Recount nodes.
+                        nodeCount = NodeTypeCountVisitor.countNodes(queryTree);
```

Comment:
Don't need this call.  Just call once after expanding regexes and ranges

Reply:
Removed.